frontend/app/src/App.tsx

```diff
@@ -371,6 +371,7 @@ export class App extends PureComponent<Props, State> {
       },
       restartWebsocketConnection: () => {
         if (!this.connectionManager) {
+          this.setState({ scriptRunState: ScriptRunState.RERUN_REQUESTED })
```

Comment:
Might put a comment that this is an intentional restart so we want the script to rerun on load (different from network connectivity issues).